### PR TITLE
Add new dependency type InterruptedBy=

### PR DESCRIPTION
**Pull request by bl33pbl0p**
Fixes: #5966 

This decouples the stop job propagation from BindsTo= for when we only want a unit to "interrupt" us through JOB_STOP propagation on abrupt transition to inactive/failed state, through a new dependency type. Combine this with Requires=/PartOf=, and you get propagation effects similar to BindsTo=.

I might be missing a few things here and there (where else to place any checks or any D-Bus stuff), but it's fairly isolated and triggers on a very specific event.

While this can be mixed-n-matched with any other dependencies, the particular usecase in the issue can be done using this setup:

```ini
#foo.service
[Unit]
Requisite=foobar.device
InterruptedBy=foobar.device
After=foobar.device
[Service]
...
```

Also, using Requires=+PartOf=+InterruptedBy= is akin to BindsTo= in terms of propagation effects.

### CODE REVIEW

File: man/systemd.unit.xml [hunk 1]
```diff
@@ -634,6 +634,16 @@
...
 
+      <varlistentry>
+	<term><varname>InterruptedBy=</varname></term>
+
+	<listitem><para>Configures propagation of unit state changes that do not go through systemd's job engine, like abrupt
+	disappearance of units (on direct transition to inactive and failed). This is mostly useful on its own but can be combined with
+	other dependency types to get roughly the same propagation effects like <varname>BindsTo=</varname>, in particular
+	<varname>Requires=</varname>, and <varname>PartOf=</varname>.</para>
+	</listitem>
+      <varlistentry>
```
Comment (mrc0mmand) [inline man/systemd.unit.xml]: Should be `</varlistentry>`

Comment (bl33pbl0p) [reply man/systemd.unit.xml]: oops

**Comment by bl33pbl0p** [comment]
ping @poettering

**Comment by poettering** [comment]
sorry for the late review, this somehow escaped my eye. 

Can you give some examples where this new type of dep is really useful?

It appears to me this is mostly useful when a device *might* be used by some service, but possibly not. But that's awfully vague, no?

A better name might be BindStopTo=…?

**Comment by poettering** [comment]
(also needs rebase)

**Comment by keszybz** [comment]
> Can you give some examples where this new type of dep is really useful?

Seconded.

@bl33pbl0p is this still something that you want to pursue?

**Comment by poettering** [comment]
hey @bl33pbl0p! @dtardon pointed out that the `PropagateStopFrom=` stuff from #19322 does pretty much what this PR adds. Right?

I guess we can close this here and continue in #19322?

**Comment by poettering** [comment]
Lets close in favour of #19322, and continue discussions there.

**Pull request closed without merging**